django/db/backends/postgresql/schema.py

```diff
@@ -111,13 +111,13 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
             new_db_params, strict,
         )
         # Added an index? Create any PostgreSQL-specific indexes.
-        if ((not old_field.db_index and new_field.db_index) or (not old_field.unique and new_field.unique)):
+        if not old_field.db_index and not old_field.unique and (new_field.db_index or new_field.unique):
```

Comment:
Oh are we creating the `_like` indexes even when `unique=True`? I wasn't expecting that.